src/leetCodeSolutionProvider.ts

```diff
@@ -0,0 +1,134 @@
+        });
+        this.markdownPath = path.join(process.env.VSCODE_CWD as string, "resources", "app", "extensions", "markdown-language-features");
+
+        // Override code_block rule for highlighting in solution language
+        // tslint:disable-next-line:typedef
+        this.markdown.renderer.rules["code_block"] = (tokens, idx, options, _, self) => {
+            const highlight: string = options.highlight(tokens[idx].content, undefined);
+            return [
+                `<pre><code ${self.renderAttrs(tokens[idx])} >`,
+                highlight || this.markdown.utils.escapeHtml(tokens[idx].content),
+                "</code></pre>",
+            ].join("\n");
+        };
+    }
+
+    public async show(solutionString: string, problem: IProblem): Promise<void> {
+        if (!this.panel) {
+            this.panel = window.createWebviewPanel("leetCode", "Top voted solution", ViewColumn.Active, {
+                retainContextWhenHidden: true,
+                enableFindWidget: true,
+                localResourceRoots: [vscode.Uri.file(path.join(this.markdownPath, "media"))],
+            });
+
+            this.panel.onDidDispose(() => {
+                this.panel = undefined;
+            }, null, this.context.subscriptions);
+        }
+
+        this.solution = this.parseSolution(solutionString);
+        this.panel.title = problem.name;
+        this.panel.webview.html = this.getWebViewContent(this.solution);
+        this.panel.reveal(ViewColumn.Active);
+    }
+
+    public dispose(): void {
+        if (this.panel) {
+            this.panel.dispose();
+        }
+    }
+
+    private parseSolution(raw: string): Solution {
+        const solution: Solution = new Solution();
+        // [^] matches everything including \n, yet can be replaced by . in ES2018's `m` flag
+        raw = raw.slice(1); // skip first empty line
+        [solution.title, raw] = raw.split(/\n\n([^]+)/); // parse title and skip one line
+        [solution.url, raw] = raw.split(/\n\n([^]+)/); // parse url and skip one line
+        [solution.lang, raw] = raw.match(/\* Lang:\s+(.+)\n([^]+)/)!.slice(1);
+        [solution.author, raw] = raw.match(/\* Author:\s+(.+)\n([^]+)/)!.slice(1);
+        [solution.votes, raw] = raw.match(/\* Votes:\s+(\d+)\n\n([^]+)/)!.slice(1);
+        solution.body = raw;
+        return solution;
+    }
+
+    private codeHighlighter(code: string, lang: string | undefined): string {
+        if (!lang) {
+            lang = this.solution.lang;
+        }
+        if (hljs.getLanguage(lang)) {
+            try {
+                return hljs.highlight(lang, code, true).value;
+            } catch (error) { /* do not highlight */ }
+        }
+        return ""; // use external default escaping
+    }
+
+    private getMarkdownStyles(): vscode.Uri[] {
+        const stylePaths: string[] = require(path.join(this.markdownPath, "package.json"))["contributes"]["markdown.previewStyles"];
```

Comment:
Emm.. Got `Cannot find module '/Applications/Visual Studio Code.app/Contents/Resources/app/resources/app/extensions/markdown-language-features/package.json'` here.

While I can confirm that this path does exist, but have no idea why it could not be found. Do you know why?

I'm thinking that at least we should add a try-catch block here and return `[]` in the catch in case that we cannot get the style paths

Reply:
No idea what happened... After I switch to `vscode.env.appRoot`, the code both works on my Win and Linux machine. 

Could you have one more try one the new commit? If the problem still exists, I'll go somewhere to find a Mac to look into the problem.

Reply:
![image](https://user-images.githubusercontent.com/20227484/54259579-3ef7f700-45a1-11e9-9e79-922f37231f21.png)
I still got the style work on a Mac...Yet it may be worth noting that there is `Users/<username>` before `Application`.
If it means that your code is installed in root applications, then does it have anything to do with the file access authorization?

Reply:
@Vigilans My fault, I forget to remove the `resources/app`, sorry for the confusion, haha